### PR TITLE
 Add bundle

### PR DESCRIPTION
**Pull request by steevanb**
Create the bundle who modify some behaviors of [trikoder/oauth2-bundle](https://github.com/trikoder/oauth2-bundle)

### CODE REVIEW

File: src/Command/CreateClientCommand.php [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+use Symfony\Component\Console\Style\SymfonyStyle;
+use TeaEbook\Oauth2BundleExtended\Entity\ClientName;
+use Trikoder\Bundle\OAuth2Bundle\Manager\ClientManagerInterface;
+use Trikoder\Bundle\OAuth2Bundle\Model\Client;
+use Trikoder\Bundle\OAuth2Bundle\Model\Grant;
+use Trikoder\Bundle\OAuth2Bundle\Model\RedirectUri;
+use Trikoder\Bundle\OAuth2Bundle\Model\Scope;
+
+/**
+ * Code copied from vendor/trikoder/oauth2-bundle/Command/CreateClientCommand.php because original class is final...
```
Comment (cGuille) [inline src/Command/CreateClientCommand.php:26]: Can we also add comments where we made changes, then?

File: src/Entity/ClientName.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    /** @var int|null */
+    protected $id;
+
+    /** @var Client|null */
+    protected $client;
+
+    /** @var string|null */
+    protected $name;
+
+    public function __construct(Client $client, string $name = null)
```
Comment (cGuille) [inline src/Entity/ClientName.php:23]: Does it actually make sense to create one without a name?

Comment (vmararunicoara) [reply src/Entity/ClientName.php:23]: As it is not mandatory I don't feel that we should force it. @steevanb other thoughts?

Comment (cGuille) [reply src/Entity/ClientName.php:23]: Oh, so it is mandatory to have a row in `oauth2_client_name` even if the client does not have a name? I think that is what confused me, but I guess that makes it consistent with the name column being optional.

Comment (steevanb) [reply src/Entity/ClientName.php:23]: I've just take the old code to move it here, i don't want to change this behavior

we have only one creation of this entity, and name is optionnal: https://github.com/TEA-ebook/oauth2-bundle-extended/blob/bundle/src/Command/CreateClientCommand.php#L189 (`$input->getOption('name')` is null if we don't add `--name=` in CLI)

**Review by cGuille — DISMISSED** [review]

**Review by vmararunicoara — DISMISSED** [review]

**Review by vmararunicoara — APPROVED** [review]

**Review by cGuille — APPROVED** [review]

**Pull request merged**